xls/ir/nodes_source.tmpl

```diff
@@ -105,6 +105,29 @@ SliceData Concat::GetOperandSliceData(int64 operandno) const {
   return SliceData{start, this->operand(operandno)->BitCountOrDie()};
 }
 
+absl::StatusOr<Node*> Param::Clone(absl::Span<Node* const> new_operands,
+                                           Function* new_function) const {
+  // TODO(meheff): Choose an appropriate name for the cloned node.
+  XLS_RET_CHECK_EQ(operand_count(), new_operands.size());
+  XLS_ASSIGN_OR_RETURN(Type* new_type, 
```

Comment:
Hmm, ick. I guess there is no way around this. We need something higher scoped to hold the types. Nothing for you to do.

Reply:
Ack.